File: frontend/src/features/skill-gap/types/skillGap.ts
export type SkillGapEvidenceSource = {
  paper_id: string;
  title: string | null;
  final_score: number;
  matched_fields: string[];
  top_contributing_signals: string[];
};

export type MissingSkillEvidence = {
  skill: string;
  confidence: number;
  evidence_count: number;
  supporting_papers: SkillGapEvidenceSource[];
  rationale: string;
};

export type SkillGapResponse = {
  user_id: string;
  current_skills: string[];
  missing_skills: MissingSkillEvidence[];
  suggested_next_skills: string[];
  strengths: string[];
  gaps_summary: string;
};

File: backend/app/models/skill_gap.py
"""Response models for backend-driven skill gap analysis."""

from __future__ import annotations

from pydantic import BaseModel, Field


class SkillGapEvidenceSource(BaseModel):
    """One recommendation item that supports a missing skill inference."""

    paper_id: str
    title: str | None = None
    final_score: float
    matched_fields: list[str] = Field(default_factory=list)
    top_contributing_signals: list[str] = Field(default_factory=list)


class MissingSkillEvidence(BaseModel):
    """Derived missing skill with confidence and supporting evidence."""

    skill: str
    confidence: float
    evidence_count: int
    supporting_papers: list[SkillGapEvidenceSource] = Field(default_factory=list)
    rationale: str


class SkillGapResponse(BaseModel):
    """Canonical skill gap response for frontend visualization."""

    user_id: str
    current_skills: list[str] = Field(default_factory=list)
    missing_skills: list[MissingSkillEvidence] = Field(default_factory=list)
    suggested_next_skills: list[str] = Field(default_factory=list)
    strengths: list[str] = Field(default_factory=list)
    gaps_summary: str
